### PR TITLE
Firmware Retract fixes and Simplify3d/Slic3r compatible

### PR DESCRIPTION
**Pull request by ghost**
**FWRETRACT** 

- Zlift Feedrate to the max 
- Allow G1 Zxx  ' z moves of the slicer ' , between zlifting
- Allowed G10S1 After G10
- G10 after G10S1 rejected
- Only One zlift/hop

**Very important feature** 
- LCD menu for swap_retract_recover_feedrate; Not translated in all language

**G10** 

- Before : Force swap statut ' big problem because must need processing before '
- Now   :  Swap statut is suggested and the firmware function will apply after processing

**Note** 

> Now compatible , with simplify3d  ' Zmoving between zlifting now possible '
> Now compatible with slic3r ' G10 after G10S1 keep the swap statut activated '

My First PR 👍

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -3205,25 +3207,32 @@ static void homeaxis(const AxisEnum axis) {
       current_position[E_AXIS] += (swapping ? swap_retract_length : retract_length) / volumetric_multiplier[active_extruder];
       sync_plan_position_e();
       prepare_move_to_destination();
-
-      // Is a Z hop set, and has the hop not yet been done?
-      if (has_zhop) {
```
Comment (ghost) [inline Marlin/Marlin_main.cpp:3210]: **No conditionnals here** 
The machine is the machine 
The gcode is the gcode
If the machine make a hop , than the machine can recover and it's all

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -117,7 +117,8 @@
  * M100 - Watch Free Memory (for debugging) (Requires M100_FREE_MEMORY_WATCHER)
  * M104 - Set extruder target temp.
  * M105 - Report current temperatures.
- * M106 - Fan on.
+ * M106 - Fan on: S0-255 : T3-255 / T2 / T1 Temporary set speed during printing
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:120]: Please move documentation like this to the `M106` handler. These descriptions are intentionally terse.

File: Marlin/Marlin_main.cpp [hunk 3]
```diff
@@ -3180,7 +3183,7 @@ static void homeaxis(const AxisEnum axis) {
       }
       SERIAL_ECHOLNPAIR("current_position[z] ", current_position[Z_AXIS]);
       SERIAL_ECHOLNPAIR("hop_amount ", hop_amount);
-    //*/
+    */
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:3186]: No one ever gets the purpose of this. Please leave these these as-is.

File: Marlin/Marlin_main.cpp [hunk 4]
```diff
@@ -3240,12 +3250,13 @@ static void homeaxis(const AxisEnum axis) {
     // The active extruder is now retracted or recovered
     retracted[active_extruder] = retracting;
 
-    // If swap retract/recover then update the retracted_swap flag too
+    // If swap retract/recover then update the retracted_swap flag too
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:3253]: No need to add trailing spaces here.

File: Marlin/Marlin_main.cpp [hunk 5]
```diff
@@ -3240,12 +3250,13 @@ static void homeaxis(const AxisEnum axis) {
...
-    // If swap retract/recover then update the retracted_swap flag too
+    // If swap retract/recover then update the retracted_swap flag too  
     #if EXTRUDERS > 1
-      if (swapping) retracted_swap[active_extruder] = retracting;
+      if (swapping) {
+        retracted_swap[active_extruder] = retracting;  }
     #endif
 
-    /* // debugging
+     /* debugging
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:3259]: (has a purpose)

File: Marlin/ultralcd.cpp [hunk 6]
```diff
@@ -3569,6 +3569,7 @@ void kill_screen(const char* lcd_msg) {
         MENU_ITEM_EDIT(float52, MSG_CONTROL_RETRACT_RECOVER_SWAP, &swap_retract_recover_length, -100, 100);
       #endif
       MENU_ITEM_EDIT(float3, MSG_CONTROL_RETRACT_RECOVERF, &retract_recover_feedrate_mm_s, 1, 999);
+      MENU_ITEM_EDIT(float3, "Lg Unret Feed", &swap_retract_recover_feedrate_mm_s, 1, 999);
```
Comment (thinkyhead) [inline Marlin/ultralcd.cpp:3572]: This requires a translatable string.

**Comment by thinkyhead** [comment]
> My First PR 👍 

Congrats! It needs a few changes before it can be accepted. I won't bother you over every missing or extra space, but I will do a thorough cleanup of the code before it gets merged.

**Pull request closed without merging**